ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -348,8 +357,13 @@ bracketChainSyncClient
         withWatcher
             "ChainSync.Client.rejectInvalidBlocks"
             (invalidBlockWatcher varState)
-      $ LeakyBucket.execAgainstBucket lopBucketConfig
-      $ \lopBucket ->
+      $ LeakyBucket.execAgainstBucket'
+      $ \lopBucket -> do
+            -- NOTE: Updating the bucket configuration and registering the
+            -- callback should be in the same STM transaction.
```

Comment:
Updating the bucket configuration is not an `STM` action, so how can it be done "in the same STM transaction" as something else? Is this about changing `updateLopBucketConfig` such that it *is* in STM?

Reply:
That is what I mean, indeed. I would like `updateLoPBucketConfig` to be in `STM` and that would allow the code here to be just one STM transaction which I find more robust.

Reply:
That is what I mean, indeed. I would like `updateLoPBucketConfig` to be in `STM` and that would allow the code here to be just one STM transaction which I find more robust.

Reply:
I tried to improve the comment in 40a04720cf92814c7bd081c9e7f0f0b6050bcef8.